Install error: variable has incomplete type 'struct stat64'

Hi, Attempting to install rpm on Big Sur with M1 chip via HomeBrew. 
Initially getting "Error: rpm: no bottle available" 

I have since tried both of the following commands: 
**brew install --build-from-source rpm** and **brew install rpm --build-bottle** 

Both give me the following error: 

> robertmurphy@ALT-RM-13205 Repositories % brew install --build-from-source rpm
> ==> Downloading http://ftp.rpm.org/releases/rpm-4.16.x/rpm-4.16.1.3.tar.bz2
> Already downloaded: /Users/robertmurphy/Library/Caches/Homebrew/downloads/490bdabcf09ad0ae5e0dc4820a86faad5112c2470621cb3153e74a4003d3776c--rpm-4.16.1.3.tar.bz2
> ==> ./configure --prefix=/opt/homebrew/Cellar/rpm/4.16.1.3 --localstatedir=/opt/homebrew/var --sharedstatedir=/opt/homebrew/var/lib --sysconfdir=/opt/homebrew/etc --with-path-magic=/o
> ==> make install
> Last 15 lines from /Users/robertmurphy/Library/Logs/Homebrew/rpm/02.make:
> /bin/sh ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H   -I.. -I.. -I../include -I../misc -I/opt/homebrew/opt/lua/include/lua -I/opt/homebrew/Cellar/openssl@1.1/1.1.1k/include  -g -O2 -c -o fts.lo fts.c
> /bin/sh ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H   -I.. -I.. -I../include -I../misc -I/opt/homebrew/opt/lua/include/lua -I/opt/homebrew/Cellar/openssl@1.1/1.1.1k/include  -g -O2 -c -o fnmatch.lo fnmatch.c
> libtool: compile:  clang -DHAVE_CONFIG_H -I.. -I.. -I../include -I../misc -I/opt/homebrew/opt/lua/include/lua -I/opt/homebrew/Cellar/openssl@1.1/1.1.1k/include -g -O2 -c fnmatch.c  -fno-common -DPIC -o .libs/fnmatch.o
> libtool: compile:  clang -DHAVE_CONFIG_H -I.. -I.. -I../include -I../misc -I/opt/homebrew/opt/lua/include/lua -I/opt/homebrew/Cellar/openssl@1.1/1.1.1k/include -g -O2 -c fts.c  -fno-common -DPIC -o .libs/fts.o
> fts.c:1120:16: error: variable has incomplete type 'struct stat64'
>         struct stat64 sb;
>                       ^
> fts.c:1120:9: note: forward declaration of 'struct stat64'
>         struct stat64 sb;
>                ^
> 1 error generated.
> make[2]: *** [fts.lo] Error 1
> make[2]: *** Waiting for unfinished jobs....
> make[1]: *** [install-recursive] Error 1
> make: *** [install] Error 2

Can anybody help me with this please? 
Let me know if any more information is required. Thanks

--- a/misc/fts.c
+++ b/misc/fts.c
@@ -32,7 +32,7 @@ static char sccsid[] = "@(#)fts.c	8.6 (Berkeley) 8/14/94";
 #endif /* LIBC_SCCS and not lint */
 
 /* Conditional to set up proper fstat64 implementation */
-#if defined(hpux) || defined(sun)
+#if defined(hpux) || defined(sun) || defined(__APPLE__)
 #   define FTS_FSTAT64(_fd, _sbp)   fstat((_fd), (_sbp))
 #else
 #   define FTS_FSTAT64(_fd, _sbp)   fstat64((_fd), (_sbp))
@@ -63,6 +63,7 @@ static char sccsid[] = "@(#)fts.c	8.6 (Berkeley) 8/14/94";
 #endif
 #if defined(__APPLE__)
 #   define __errno_location()	(__error())
+#   define stat64		stat
 #endif
 
 #include "system.h"

